colossalai/inference/tensor_parallel/modeling/_utils.py

```diff
@@ -1,10 +1,51 @@
 """
 Utils for model inference
 """
+import os
+
+import torch
+
 from colossalai.kernel.triton.copy_kv_cache_dest import copy_kv_cache_to_dest
 
 
 def copy_kv_to_mem_cache(layer_id, key_buffer, value_buffer, context_mem_index, mem_manager):
     copy_kv_cache_to_dest(key_buffer, context_mem_index, mem_manager.key_buffer[layer_id])
     copy_kv_cache_to_dest(value_buffer, context_mem_index, mem_manager.value_buffer[layer_id])
     return
+
+
+def init_to_get_rotary(self, base=10000, use_elem=False):
```

Comment:
docstring is required to explain this function, e.g. what does it do, what models can it apply to, etc.

Reply:
for sure!

Reply:
done